liblwgeom/lwgeom_geos.c

```diff
@@ -443,19 +443,21 @@ LWGEOM2GEOS(const LWGEOM *lwgeom, int autofix)
 		ngeoms = lwc->ngeoms;
 		if ( ngeoms > 0 )
 			geoms = malloc(sizeof(GEOSGeom)*ngeoms);
-
+		j = 0;
+		// i is index of lwgeom geometry, j is index of geos geometry - skip empty collection entries
 		for (i=0; i<ngeoms; ++i)
 		{
+			if (lwgeom_is_empty(lwgeom)) continue; // skip empty
```

Comment:
To skip empty sub-geometries, shouldn't this be
```c
if(lwgeom_is_empty(lwc->geoms[i]))
```
right now this is just skipping only in the case where the whole collection is empty, which I'm surprised isn't handled further up the chain